src/runner.py

```diff
@@ -248,7 +248,22 @@ def _create_instance(
             "profiles": ["default", "runner", resource_profile],
         }
 
-        instance = self._clients.lxd.instances.create(config=instance_config, wait=True)
+        try:
+            instance = self._clients.lxd.instances.create(config=instance_config, wait=True)
+        except LxdError:
+            logger.exception(
+                "Removing resource profile and storage profile due to LXD instance create failure"
+            )
+
+            # LxdError on creating LXD instance could be caused by improper initialization of
+            # storage pool. If other runner LXD instance exists then it cannot be the cause.
+            if self._clients.lxd.instances.all():
```

Comment:
When I read the comment I would expect a `not self._clients.lxd.instances.all()` .